src/store/app.ts

```diff
@@ -301,5 +382,136 @@ export const useAppStore = defineStore('app', {
         popAllNotifications(): Notification[] {
             return this.queuedNotifications.splice(0)
         },
+        /**
+         * Asynchronously adds a product variant to the shopping cart.
+         * If the product has already been added to the cart
+         * then it updates the count:
+         * It simply adds the specified count on top of the existing count.
+         *
+         * @param productVariantId - The ID of the product variant to add.
+         * @param count - The quantity of the product variant to add.
+         */
+        async addProductVariantToShoppingCart(
```

Comment:
same as above, maybe refactoring into multiple functions is reasonably possible

Reply:
89fda6b208dbabdbc3f6e13b391b53f5b1e889ab